submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,164 @@
+      };
+    };
+    
+    shuffle(numbersArray);
+    
+    if (document.documentElement.clientHeight > document.documentElement.clientWidth) {
+      if (sizeSelector.value == 12) {
+        gameArea.classList.add("game_area-3columns");
+      } else {
+        gameArea.classList.add("game_area-4columns");
+      };
+    } else {
+      if (sizeSelector.value <= 16) {
+        gameArea.classList.add("game_area-4columns");
+      } else if (sizeSelector.value == 20){
+        gameArea.classList.add("game_area-5columns");
+      } else {
+        gameArea.classList.add("game_area-6columns");
+      };
+    };
+    let fragment = document.createDocumentFragment();
+    for (let i=0; i < sizeSelector.value; i++) {
+      let card = document.createElement("button");
+      card.classList.add("card");
+      card.innerHTML = `<div class="card__front" style="background-image: url(img/${cardThemeSelected.value}_bg.jpg);"></div><div class="card__back" style="background-image: url(img/${cardThemeSelected.value}_${numbersArray[i] + 1}.jpg);"></div>`;
+      fragment.append(card);
+      card.addEventListener("click", checkCard);
+    };
+    gameArea.append(fragment);
+    cardsArray = Array.from(document.querySelectorAll(".card"));
+  };
+  
+  function checkCard({target}) {
+    flipCard.call(target);
+    setTimeout(function() {
+      if (!cardSelected) {
+        cardSelected = cardsArray.indexOf(target) + 1;
+        lockCard.call(target);
+      } else if (numbersArray[cardSelected - 1] == numbersArray[cardsArray.indexOf(target)]) {
+        vanishCard.call(cardsArray[cardSelected - 1]);
+        vanishCard.call(target);
+        cardSelected = "";
+        matchedCounter++;
+        if (matchedCounter == sizeSelector.value/2) {
+          hailWinner();
+        };
+      } else {
+        flipCard.call(cardsArray[cardSelected - 1]);
+        unlockCard.call(cardsArray[cardSelected - 1])
+        cardSelected = "";      
+        flipCard.call(target);
+      };
+    }, 400);
+    
+    function vanishCard() {
+      this.classList.add("card-vanish");
+      this.setAttribute("tabindex", "-1");
+    };
+    
+    function flipCard() {
+      this.firstElementChild.classList.toggle("card__front-flip");
+      this.lastElementChild.classList.toggle("card__back-flip");
+    };
+    
+    function lockCard() {
+      this.style.pointerEvents = "none";
+      this.setAttribute("disabled", "true");
+    };
+    
+    function unlockCard() {
+      this.style.pointerEvents = "auto";
+      this.removeAttribute("disabled");
+    };
+  };
+
+  function hailWinner() {
+    const hailWindow = document.createElement("div");
+    hailWindow.innerHTML = `Вы выиграли!<button class="popup_hail__button">Ok</button>`;
+    hailWindow.classList.add("popup_hail");
+    doc.append(hailWindow);
+    const okButton = document.querySelector(".popup_hail__button");
+    okButton.addEventListener("click", function () {
+      document.location.reload();
```

Comment:
Maybe better clean the game area and draw cards again

Reply:
There are two ways: long (and "right") way - clean the game area and enable game selectors in HEADER, and short (and quick) way - reload the page. I chose short way to avoid accusation in un-optimality :)

Reply:
In this case, you will have to re-select the themes, the hero, and the size of the playing field (

Reply:
Well, when page reloads JS runs again, and so all the elements are being selected again.
But I will remake this, I will not be lazy :) and make it the "right" (and long) way.

Reply:
I remade "final" window in the "right" way.